### ISSUE TITLE
Rule S3400: FP when using default interface implementation

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#5498]
### Description

S3400 raises an issue for default interface implementation. The method can be overridden so the behavior should be similar with virtual methods.

Source: https://community.sonarsource.com/t/csharpsquid-s3400-should-not-apply-to-default-interface-implementations/60075

### Repro steps

```
interface ISomeInterface {
    public bool IsCondition() => true; // this triggers the rule
}

class SomeClass implements ISomeInterface {
    public bool Value { get; set; }
    public bool IsCondition() => Value; // this is not a constant
}
```

### Expected behavior

The rule should not be raised.

### Actual behavior

The `IsCondition` method is considered noncompliant.

### Known workarounds

Suppress the raised warning.

### Related information

* C#/VB.NET Plugins version: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.36.1.44192
